packages/syft/src/syft/core/node/common/action/save_object_action.py

```diff
@@ -43,6 +43,17 @@ def __repr__(self) -> str:
         return f"SaveObjectAction {obj_str}"
 
     def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
+        # Check if it uses an id previously registered.
+        old_obj = node.store.get_or_none(key=self.obj.id, proxy_only=True)
+
+        if old_obj:
+            # Check if users' verify key is a subset of write_permissions set
```

Comment:
MAYBE:  We could be a little more defensive , we could prevent clients or even the root users from writing to the same UID
https://github.com/OpenMined/PySyft/issues/6965#issuecomment-1306553924
Since we also have the mutation problem where we overwrite the same UID.
https://github.com/OpenMined/PySyft/issues/6976
We could allow admin users to overwrite particular UID, no particular use-case comes to mind, where this would be useful.
Thoughts?